addons/mail/models/mail_message.py

```diff
@@ -872,7 +872,13 @@ def _message_reaction(self, content, action):
         group_domain = [("message_id", "=", self.id), ("content", "=", content)]
         count = self.env["mail.message.reaction"].search_count(group_domain)
         group_command = "ADD" if count > 0 else "DELETE"
-        personas = [("ADD" if action == "add" else "DELETE", {"id": guest.id if guest else partner.id, "type": "guest" if guest else "partner"})] if guest or partner else []
+        persona = guest or partner
+        personas = []
+        if persona:
```

Comment:
As you commented on portal chatter PR ([raise `NotFound` if there is no guest or partner](https://github.com/odoo/odoo/pull/138233#discussion_r1729180665)), isn't better to raise `AccessError` here at the first steps of this method?

Reply:
I don't think it's possible to reach this code without either a guest or a partner in v17/without chatter portal with token?

Either way I would not change that in stable, just keeping the same behavior as before the PR and adding the missing fields.